go/vt/vtgate/planbuilder/operators/update.go

```diff
@@ -322,31 +394,36 @@ func createFkChildForUpdate(ctx *plancontext.PlanningContext, fk vindexes.ChildF
 	}
 
 	return &FkChild{
-		BVName: bvName,
-		Cols:   cols,
-		Op:     childOp,
+		BVName:         bvName,
+		Cols:           selectOffsets,
+		Op:             childOp,
+		NonLiteralInfo: nonLiteralUpdateInfo,
 	}, nil
 }
 
 // buildChildUpdOpForCascade builds the child update statement operator for the CASCADE type foreign key constraint.
 // The query looks like this -
 //
 //	`UPDATE <child_table> SET <child_column_updated_using_update_exprs_from_parent_update_query> WHERE <child_columns_in_fk> IN (<bind variable for the output from SELECT>)`
-func buildChildUpdOpForCascade(ctx *plancontext.PlanningContext, fk vindexes.ChildFKInfo, updStmt *sqlparser.Update, childWhereExpr sqlparser.Expr, updatedTable *vindexes.Table) (ops.Operator, error) {
+func buildChildUpdOpForCascade(ctx *plancontext.PlanningContext, fk vindexes.ChildFKInfo, childWhereExpr sqlparser.Expr, nonLiteralUpdateInfo []engine.NonLiteralUpdateInfo, updatedTable *vindexes.Table) (ops.Operator, error) {
```

Comment:
nitpick: these lines are really long. shouldn't we split lines when they grow too long?

Reply:
Done